File: backend/restaurantapp/src/main/java/com/foodapp/restrauentapp/model/Restaurant.java
package com.foodapp.restrauentapp.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data

@Document
public class Restaurant {

    @Id
    private String id;
    private String name;
    private String contact;
    private String userId;
    private String cuisineType;
    private String description;
    private double rating;
    private Owner owner;
    private Address address;


}

File: frontend/FoodHut/src/app/model/Restaurant.ts
import { Address } from "./Address"
import { Owner } from "./Owner"


export type Restaurant = {
    id?:string,
    name?:string,
    contact?:string,
    userId?:string,
    cuisineType?:string,
    description?:string,
    rating?:number,
    owner?:Owner,
    address?:Address,
}